tests/v2/integration/catalogv2/system_charts_test.go

```diff
@@ -0,0 +1,439 @@
+	w.T().Helper()
+	w.Require().NoError(w.resetManagementCluster())
+	w.Require().NoError(w.uninstallApp("cattle-system", "rancher-aks-operator"))
+	w.Require().NoError(w.uninstallApp("cattle-system", "rancher-aks-operator-crd"))
+	clusterRepo, err := w.catalogClient.ClusterRepos().Get(context.TODO(), "rancher-charts", metav1.GetOptions{})
+	w.Require().NoError(err)
+	if clusterRepo.Spec.GitRepo != "https://github.com/rancher/charts" || clusterRepo.Spec.GitBranch != "release-v2.7" {
+		clusterRepo.Spec.GitRepo = "https://github.com/rancher/charts"
+		clusterRepo.Spec.GitBranch = "release-v2.7"
+		clusterRepo, err = w.catalogClient.ClusterRepos().Update(context.TODO(), clusterRepo, metav1.UpdateOptions{})
+		w.Require().NoError(err)
+		downloadTime := clusterRepo.Status.DownloadTime
+		w.Require().NoError(w.pollUntilDownloaded("rancher-charts", downloadTime))
+	}
+}
+
+func TestOperatorChartsSuite(t *testing.T) {
+	suite.Run(t, new(SystemChartsTest))
+}
+
+func (w *SystemChartsTest) TestInstallChartLatestVersion() {
+	defer w.resetSettings()
+
+	w.Require().NoError(w.updateManagementCluster())
+	app, err := w.waitForAksChart(rv1.StatusDeployed, "rancher-aks-operator", 0)
+	w.Require().NoError(err)
+	latest, err := w.catalogClient.GetLatestChartVersion("rancher-aks-operator")
+	w.Require().NoError(err)
+	w.Assert().Equal(app.Spec.Chart.Metadata.Version, latest)
+}
+
+func (w *SystemChartsTest) TestUpgradeChartToLatestVersion() {
+	defer w.resetSettings()
+
+	clusterRepo, err := w.catalogClient.ClusterRepos().Get(context.TODO(), "rancher-charts", metav1.GetOptions{})
+	w.Require().NoError(err)
+	origClusterRepo := clusterRepo.DeepCopy()
+	cfg, err := w.client.Steve.SteveType(configmaps.ConfigMapSteveType).ByID(fmt.Sprintf("%s/%s", origClusterRepo.Status.IndexConfigMapNamespace, origClusterRepo.Status.IndexConfigMapName))
+	w.Require().NoError(err)
+	cfgMap := &v1.ConfigMap{}
+	w.Require().NoError(stevev1.ConvertToK8sType(cfg.JSONResp, cfgMap))
+	origCfg := cfgMap.DeepCopy()
+
+	// GETTING INDEX FROM CONFIGMAP AND MODIFYING iT
+	gz, err := gzip.NewReader(bytes.NewBuffer(cfgMap.BinaryData["content"]))
+	w.Require().NoError(err)
+	defer gz.Close()
+	data, err := io.ReadAll(gz)
+	w.Require().NoError(err)
+	index := &repo.IndexFile{}
+	w.Require().NoError(json.Unmarshal(data, index))
+	index.SortEntries()
+	latestVersion := index.Entries["rancher-aks-operator"][0].Version
+	index.Entries["rancher-aks-operator"] = index.Entries["rancher-aks-operator"][1:]
+	//index.Entries["rancher-aks-operator-crd"] = index.Entries["rancher-aks-operator-crd"][1:]
+	marshal, err := json.Marshal(index)
+	w.Require().NoError(err)
+	var compressedData bytes.Buffer
+	writer := gzip.NewWriter(&compressedData)
+	_, err = writer.Write(marshal)
+	w.Require().NoError(err)
+	w.Require().NoError(writer.Close())
+	cfgMap.BinaryData["content"] = compressedData.Bytes()
+
+	//UPDATING THE CONFIGMAP
+	cfg, err = w.client.Steve.SteveType(configmaps.ConfigMapSteveType).Update(cfg, cfgMap)
```

Comment:
can we use client like catalogclient to fetch configmap. ? Clients are like abstraction for api since api can change